### DISCUSSION TITLE
How do I filter and modify waiting tasks?

### DISCUSSION THREAD

**Discussion by gevhaz** [GothenburgBitFactory/taskwarrior#2456] [Community support] [answered]
Hello, thanks for developing taskwarrior. It's super useful.

I'm probably missing something obvious but I wonder how I would go about accessing a subset of waiting tasks and modify them. I want do to something like:

```
task +WAITING and pro:project1 modify wait:
```

That is, filter out waiting tasks that belong to project1 and remove their waiting status.

What this actually does is to only filter due tasks, none of which of course have the waiting status. If I use `task all`, I am blocked from using `modify`. How do I get around this?

**Comment by tbabej** [comment, marked as answer]
Hi @gevhaz,

the `modify` command has no default filter, in other words `task <filter> modify` acts on the same set of tasks that `task <filter> all` displays.

Given that, the command that you shared should accomplish exactly what you describe - remove wait date from all the tasks in project1. Can you confirm this is not the case for you? Would love to get a reproducer here.

**Comment by gevhaz** [comment]
Ah, I see, it actually works. Thanks.

The reason I was confused was that I tried to run the filter without the modify part first, to make sure I was modifying the right tasks. In that case (just running `task +WAITING and pro:project1`), I'm told: 

```
[task next ( +WAITING and pro:project1 )]
No matches.
```

**Reply by tbabej** [reply]
I see where you're coming from. The culprit here is the implicit filter of the `next` report, that is getting combined with your user-provided filter:

```
[warrior@tw-2.6.0 ~]$ task show report.next  

Config Variable         Value                                                                                                                          
report.next.columns     id,start.age,entry.age,depends,priority,project,tags,recur,scheduled.countdown,due.relative,until.remaining,description,urgency
report.next.description Most urgent tasks
report.next.filter      status:pending limit:page                                                                                                      
report.next.labels      ID,Active,Age,Deps,P,Project,Tag,Recur,S,Due,Until,Description,Urg
report.next.sort        urgency-                                                                                                                       
```

You can see that the default filter contains `status:pending`, which filters out all of your matches, because presumably, these are deleted or completed tasks we're talking about here.

FYI, on 2.6.0 this is a bit more obvious with the header saying:

```
[warrior@tw-2.6.0 ~]$ task +WAITING and pro:project1 next
Filter: ( status = pending and limit:page ) and ( tags _hastag_ WAITING and pro = project1 )
No matches.
```
